modules/m_whois.c

```diff
@@ -373,10 +375,20 @@ single_whois(struct Client *source_p, struct Client *target_p, int operspy)
 					target_p->name, buf);
 		}
 
+		/* fire the doing_whois_show_idle hook to allow modules to tell us whether to show the idle time */
+		hook_data_client_approval hdata_showidle;
+
+		hdata_showidle.client = source_p;
+		hdata_showidle.target = target_p;
+		hdata_showidle.approved = 1;
+
+		call_hook(doing_whois_show_idle_hook, &hdata_showidle);
+
+		/* operspies are always allowed to see idle time */
 		sendto_one_numeric(source_p, RPL_WHOISIDLE, form_str(RPL_WHOISIDLE),
-				   target_p->name,
-				   (long)(rb_current_time() - target_p->localClient->last),
-				   (unsigned long)target_p->localClient->firsttime);
+			   target_p->name,
+			   hdata_showidle.approved || operspy ? (long)(rb_current_time() - target_p->localClient->lasttime) : 0,
```

Comment:
I disagree with including operspy here; there's already a separate priv for it, if a network wants operspy folk to see it too, they can give that priv as well.

Also, this needs to stay using last, not lasttime.  Last is the last time user sent a PRIVMSG, lasttime is the last time the user sent any line, and is only used for ping calculations.

```suggestion
			   hdata_showidle.approved ? (long)(rb_current_time() - target_p->localClient->last) : 0,
```

This suggestion is just the last revert, in case Libera disagrees with me about not including operspy here

```suggestion
			   hdata_showidle.approved || operspy ? (long)(rb_current_time() - target_p->localClient->last) : 0,
```

Reply:
After discussion on IRC, we've decided to drop the operspy bit here.